main.js

```diff
@@ -0,0 +1,54 @@
+// DOM utility functions
+const addClass = (el, newClass) => {
+  el.className += ` ${newClass}`;
+};
+
+const removeClass = (el, oldClass) => {
+  const currentClasses = el.className.split(" ");
+
+  el.className = currentClasses.filter(names =>
+    names.trim() !== "" && names !== oldClass
+  ).join(" ");
+};
+
+// Search Script
+(() => {
+  const inputField = document.getElementsByTagName('input')[0];
+  let typeTimeout;
+  const TYPE_INTERVAL = 10000;
+
+  inputField.addEventListener("input", (event) => {
+    const value = event.target.value;
+    if (typeTimeout) window.clearTimeout(typeTimeout);
+
+    if (value) {
+      typeTimeout = delay(matchKeyword(value), TYPE_INTERVAL);
+    }
+  }, false);
+
+  const matchKeyword = (value) => {
+    let matchedEmojis;
+    const matchedObjects = EMOJIS.filter(filterByKeyword.bind(this, value));
+    if (matchedObjects.length == 0) return;
+
+    if (matchedObjects.length > 1) {
+      matchedEmojis = matchedObjects.map((obj) => {
+        return obj.emoji;
+      });
+    } else {
+      matchedEmojis = matchedObjects[0].emoji;
+    }
+    // console.log(matchedEmojis);
+    return matchedEmojis;
+  }
+
+  const filterByKeyword = (value, obj) => {
+    return obj.keywords.some((keyword) => {
+      return value === keyword;
+    });
+  }
```

Comment:
> some() executes the callback function once for each element present in the array until it finds one where callback returns a truthy value (a value that becomes true when converted to a Boolean).
(from MDN)
It probably means that this is not optimal - it's not the most performant way and could use improving so that the whole "algorithm" runs faster. But for the sake of demonstrating a couple of methods at the meetup, might be ok. What do you think?